### PR TITLE
Fix HeaderValue throwing an exception on legal characters

### PR DESCRIPTION
**Pull request by avasa**
Original pull request [#24](https://github.com/zendframework/zend-mail/pull/28) created Oct 7 2015.

Author of pull request has not responded since Nov 4 2015.

Can we please get this merged and resolved.

Thanks.

### CODE REVIEW

**Comment by Maks3w** [comment]
Need tests

**Comment by avasa** [comment]
Any updates on this?

File: test/Header/HeaderValueTest.php [hunk 1]
```diff
@@ -56,7 +57,11 @@ public function validateValues()
             ["This is a\r\r test", 'assertFalse'],
             ["This is a \r\r\n test", 'assertFalse'],
             ["This is a \r\n\r\ntest", 'assertFalse'],
-            ["This is a \r\n\n\r\n test", 'assertFalse']
+            ["This is a \r\n\n\r\n test", 'assertFalse'],
+            ["This\tis\ta test", 'assertTrue'],
+            ["This is\ta \r\n test", 'assertTrue'],
+            ["This\tis\ta\ntest", 'assertFalse'],
+            ["This is a \r\t\n \r\n test", 'assertFalse'],
```
Comment (Maks3w) [inline test/Header/HeaderValueTest.php]: Please add a test case for this case `["This is a\r\n\ttest", 'asserTrue'],`

Comment (avasa) [inline test/Header/HeaderValueTest.php]: I couldn't make it assert true. Breaks other the tests. Can you please take a look. Much appreciated.

Comment (Maks3w) [inline test/Header/HeaderValueTest.php]: It's not admisible push a commit with a wrong assertion just for to make the test to pass.

We prefer a failure in the test before to ship buggy code.

Please change the assertion.

**Comment by Maks3w** [comment]
Now take a look to the line ~88 ` if ($lf !== 10 || $sp !== 32) {` play with $sp var

**Pull request closed without merging**